client/src/main/java/io/confluent/kafka/schemaregistry/avro/AvroSchemaProvider.java

```diff
@@ -15,15 +15,24 @@
 
 package io.confluent.kafka.schemaregistry.avro;
 
-import org.apache.avro.Schema;
+import org.apache.avro.SchemaParseException;
 
-public class AvroSchema {
+import io.confluent.kafka.schemaregistry.ParsedSchema;
+import io.confluent.kafka.schemaregistry.SchemaProvider;
 
-  public final Schema schemaObj;
-  public final String canonicalString;
+public class AvroSchemaProvider implements SchemaProvider {
 
-  public AvroSchema(Schema schemaObj, String canonicalString) {
-    this.schemaObj = schemaObj;
-    this.canonicalString = canonicalString;
+  @Override
+  public String schemaType() {
+    return AvroSchema.AVRO;
+  }
+
+  @Override
+  public ParsedSchema parseSchema(String schemaString) {
+    try {
```

Comment:
Related to the other comment on `AvroUtils`. Alternatively, could we call `AvroUtils.parseSchema` here to avoid duplicated code?